lib/state/data/reducer.ts

```diff
@@ -443,6 +443,17 @@ export const tags: A.Reducer<Map<T.TagHash, T.Tag>> = (
       return next;
     }
 
+    case 'RESTORE_TAGS': {
+      const next = new Map(state);
+      action.tags.forEach((tagHash) => {
+        if (!next.has(tagHash)) {
+          next.set(tagHash, { name: tagNameOf(tagHash) });
+        }
+      });
```

Comment:
see above comment: if this were watching for `RESTORE_NOTE_REVISION` then this code wouldn't need to change at all (other than `action.tags` probably becomes `action.note.tags`) and all the code in `data/middleware` doesn't need to exist.

Reply:
Yeah, I totally agree. In this [commit](https://github.com/Automattic/simplenote-electron/pull/2817/commits/728e23fbbb743c472de0118744014d4281d97e47), I'm watching for `RESTORE_NOTE_REVISION` instead.